Update What's New documentation for .NET 10 RC 2 (#49283)
dotnet/docs: docs/core/whats-new/dotnet-10/sdk.md

}
```

> ## Use .NET MSBuild Tasks with .NET Framework MSBuild

Reviewer: ```suggestion
## Use .NET MSBuild tasks with .NET Framework MSBuild
```

> MSBuild is the underlying build system for .NET, driving both build of projects (as seen in commands like `dotnet build` and `dotnet pack`), as well as acting as a general provider of information about projects (as seen in commands like `dotnet list package`, and implicitly used by commands like `dotnet run` to discover how a project wants to be executed).

Reviewer 2: ```suggestion
MSBuild is the underlying build system for .NET, driving both build of projects (as seen in commands like `dotnet build` and `dotnet pack`), and acting as a general provider of information about projects (as seen in commands like `dotnet list package`, and implicitly used by commands like `dotnet run` to discover how a project wants to be executed).
```

### Configure .NET Tasks

For task authors, opting in to this new behavior should be pretty simple—all it should take is changing your `UsingTask` declaration to tell MSBuild about your task.

```xml
<UsingTask TaskName="MyTask"
    AssemblyFile="path\to\MyTask.dll"
    Runtime="NET"
    TaskFactory="TaskHostFactory"
/>
```

The `Runtime="NET"` and `TaskFactory="TaskHostFactory"` attributes tell the MSBuild engine how to run the Task:

> - `Runtime="NET"` tells MSBuild that the Task is built for .NET (as opposed to .NET Framework)

Reviewer: ```suggestion
- `Runtime="NET"` tells MSBuild that the Task is built for .NET (as opposed to .NET Framework).
```

- `TaskFactory="TaskHostFactory"` tells MSBuild to use the `TaskHostFactory` to run the Task, which is an existing capability of MSBuild that allows tasks to be run out-of-process.

### Caveats and performance tuning

The preceding example is the simplest way to get started using .NET Tasks in MSBuild, but it has some limitations—because the `TaskHostFactory` always runs tasks out-of-process, the new .NET Task always runs in a separate process from MSBuild. This means that there is some minor overhead to running the Task because the MSBuild engine and the Task communicate over inter-process communication (IPC) instead of in-process communication. For most tasks, this overhead is negligible, but for tasks that are run many times in a build, or that do quite a lot of logging, this overhead might be more significant.

> With just a bit more work, the Task can be configured to still run in-process when running via `dotnet`:

Reviewer: ```suggestion
With just a bit more work, you can configure the task to still run in-process when running via `dotnet`:
```

    AssemblyFile="path\to\MyTask.dll"
    Runtime="NET"
    TaskFactory="TaskHostFactory"
    Condition="$(MSBuildRuntimeType) == 'Full'"
/>
<UsingTask TaskName="MyTask"
    AssemblyFile="path\to\MyTask.dll"
    Runtime="NET"
    Condition="$(MSBuildRuntimeType) == 'Core'"
/>
```

Thanks to the `Condition` feature of MSBuild, you can load a Task differently depending on whether MSBuild is running in .NET Framework (Visual Studio or `msbuild.exe`) or .NET (the `dotnet` CLI). In this example, the Task runs out-of-process when running in Visual Studio or `msbuild.exe`, but runs in-process when running in the `dotnet` CLI. This gives the best performance when running in the `dotnet` CLI, while still allowing the Task to be used in Visual Studio and `msbuild.exe`.

There are also small technical limitations to be aware of when using .NET Tasks in MSBuild—the most notable of which is that the `Host Object` feature of MSBuild Tasks is not yet supported for .NET Tasks running out-of-process. This means that if your Task relies on a Host Object, it won't work when running in Visual Studio or `msbuild.exe`. Additional support for Host Objects is planned in future releases.

> ### Future work

Reviewer: I would leave this section out. Docs aren't supposed to talk about future plans.

This is the first step in a longer journey to make MSBuild more flexible and capable. In future releases, additional capabilities will be added to MSBuild to make it easier to write and use .NET Tasks, including:

- Automatically discovering and loading .NET Tasks without needing to specify `Runtime` or `TaskFactory` metadata

> - Reducing the performance overhead of IPC between the MSBuild Engine and the Tasks when running out-of-process

Reviewer 2: ```suggestion
- Reducing the performance overhead of IPC between the MSBuild Engine and the Tasks when running out-of-process.
```

> - Supporting the `Host Object` feature for .NET Tasks running out-of-process

Reviewer 2: ```suggestion
- Supporting the `Host Object` feature for .NET Tasks running out-of-process.
```

> Task Authors can try this and provide feedback at [dotnet/msbuild](https://github.com/dotnet/msbuild/issues/new).

Reviewer 2: ```suggestion
Task authors can try this and provide feedback at [dotnet/msbuild](https://github.com/dotnet/msbuild/issues/new).
```
